src/main/java/org/folio/rest/impl/ItemStorageAPI.java

```diff
@@ -55,8 +60,20 @@ public void postItemStorageItems(
     if (entity.getStatus() == null) {
       entity.setStatus(new Status().withName(DEFAULT_STATUS_NAME));
     }
+
     PgUtil.post(ITEM_TABLE, entity, okapiHeaders, vertxContext,
-        PostItemStorageItemsResponse.class, asyncResultHandler);
+      PostItemStorageItemsResponse.class, postResponse -> {
+        // Have to re-read item to get calculated fields like effectiveLocationId
+        if (hasCreatedStatus(postResponse.result())) {
+          readItemById(entity.getId(), okapiHeaders, vertxContext)
```

Comment:
How do you know there is a need to copy the headers?

Is this code based upon a similar example elsewhere? 

Should we request that the RAML Module Builder pgUtil responds with the record from the database rather than what was provided?

Reply:
@marcjohnson-kint,

> How do you know there is a need to copy the headers?
 PgUtil.post method sets Location header internally, just checked code of this method.

> Should we request that the RAML Module Builder pgUtil responds with the record from the database rather than what was provided?
It would be great I think.